google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -179,12 +187,26 @@ public boolean value(VirtualFile file) {
         GctBundle.message("appengine.dockerfile.location.browse.button"),
         null,
         project,
-        FileChooserDescriptorFactory.createSingleFileDescriptor());
+        FileChooserDescriptorFactory
+            .createSingleFileDescriptor().withFileFilter(new Condition<VirtualFile>() {
+          @Override
+          public boolean value(VirtualFile file) {
+            return Comparing.equal(file.getName(), "Dockerfile", false);
+          }
+        })
+    );
     appYamlPathField.addBrowseFolderListener(
         GctBundle.message("appengine.appyaml.location.browse.window.title"),
         null,
         project,
-        FileChooserDescriptorFactory.createSingleFileDescriptor());
+        FileChooserDescriptorFactory
+            .createSingleFileDescriptor().withFileFilter(new Condition<VirtualFile>() {
+          @Override
+          public boolean value(VirtualFile file) {
+            return Comparing.equal(file.getName(), "app.yaml", false);
```

Comment:
Yea I was thinking about that - my original reasoning is that we would guide them on the best practice. But you give a good example here. We may want to table this PR for now.

It's worth noting that regardless of the naming of the file they choose via the IJ plugin, it will be copied over to the target directory with the canonical names (so they will always be passed to gcloud this way). This doesn't affect your point though.